d2ir/d2ir.go

```diff
@@ -751,7 +765,33 @@ func (m *Map) DeleteField(ida ...string) *Field {
 			continue
 		}
 		if len(rest) == 0 {
+			for _, fr := range f.References {
+				for i, e := range m.Edges {
+					for _, er := range e.References {
+						if er.Context == fr.Context {
+							m.DeleteEdge(e.ID)
+							i--
```

Comment:
this doesn't work in range loops. `i` will be set on the next iteration as if `i--` never happened.

Reply:
oh right. it wasn't necessary anyway, for the same reason